### PR TITLE
Fix crash with new Multi-speed Fans for WSHP Zone Equipment

### PR DESCRIPTION
**Pull request by rraustad**
Pull request overview
---------------------
 - Follow up to #9746
 - User file crashed with new code changes for Multi-speed Fans for zone equipment ([in-23.2.idf.txt](https://github.com/NREL/EnergyPlus/files/12744863/in-23.2.idf.txt))

**NOTE: ENHANCEMENTS MUST FOLLOW A SUBMISSION PROCESS INCLUDING A FEATURE PROPOSAL AND DESIGN DOCUMENT PRIOR TO SUBMITTING CODE**

### Pull Request Author
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [x] Label the PR with at least one of: Defect, Refactoring, NewFeature, Performance, and/or DoNoPublish
 - [ ] Pull requests that impact EnergyPlus code must also include unit tests to cover enhancement or defect repair
 - [x] Author should provide a "walkthrough" of relevant code changes using a GitHub code review comment process
 - [ ] If any diffs are expected, author must demonstrate they are justified using plots and descriptions
 - [ ] If changes fix a defect, the fix should be demonstrated in plots and descriptions
 - [ ] If any defect files are updated to a more recent version, upload new versions here or on DevSupport
 - [ ] If IDD requires transition, transition source, rules, ExpandObjects, and IDFs must be updated, and add IDDChange label
 - [ ] If structural output changes, add to output rules file and add OutputChange label
 - [ ] If adding/removing any LaTeX docs or figures, update that document's CMakeLists file dependencies

### Reviewer
This will not be exhaustively relevant to every PR.
 - [ ] Perform a Code Review on GitHub
 - [ ] If branch is behind develop, merge develop and build locally to check for side effects of the merge
 - [ ] If defect, verify by running develop branch and reproducing defect, then running PR and reproducing fix
 - [ ] If feature, test running new feature, try creative ways to break it
 - [ ] CI status: all green or justified
 - [ ] Check that performance is not impacted (CI Linux results include performance check)
 - [ ] Run Unit Test(s) locally
 - [ ] Check any new function arguments for performance impacts
 - [ ] Verify IDF naming conventions and styles, memos and notes and defaults
 - [ ] If new idf included, locally check the err file and other outputs

### CODE REVIEW

File: src/EnergyPlus/HVACFan.hh [hunk 1]
```diff
@@ -142,6 +142,7 @@ namespace HVACFan {
         bool AirPathFlag;                       // Yes, this fan is a part of airpath
         int m_numSpeeds;                        // input for how many speed levels for discrete fan
         std::vector<Real64> m_massFlowAtSpeed;
+        std::vector<Real64> m_flowFractionAtSpeed; // array of flow fractions for speed levels
```
Comment (rraustad) [inline src/EnergyPlus/HVACFan.hh:145]: moved to public access

File: src/EnergyPlus/UnitarySystem.cc [hunk 2]
```diff
@@ -6791,55 +6791,72 @@ namespace UnitarySystems {
                 if (state.dataHVACFan->fanObjs[FanIndex]->speedControl == HVACFan::FanSystem::SpeedControlMethod::Discrete) {
                     int NumSpeeds = state.dataHVACFan->fanObjs[FanIndex]->m_numSpeeds;
                     if (NumSpeeds > 1) {
```
Comment (rraustad) [inline src/EnergyPlus/UnitarySystem.cc:6793]: Made the following very targeted to these coil types, only for WSHP, only if air flow is not autoszied for both WSHP parent and accompanied fan object. Also, mass flow should not need to be set here but I left those lines of code, mass flow should get overwritten later unless that is somehow protected. Notice here too that the fan's m_flowFractionAtSpeed had to be made public.

**Comment by rraustad** [comment]
Defect file results in this branch (not sure what warnings were present in 23.1):

    ************* EnergyPlus Warmup Error Summary. During Warmup: 0 Warning; 0 Severe Errors.
    ************* EnergyPlus Sizing Error Summary. During Sizing: 0 Warning; 0 Severe Errors.
    ************* EnergyPlus Completed Successfully-- 53650 Warning; 0 Severe Errors; Elapsed Time=00hr 00min  5.79sec

**Comment by Myoldmopar** [comment]
Thanks @rraustad !! If CI is happy, I'm merging this and the miktex fix and taking rc2 either later tonight or in the morning. That _should_ do it.

**Comment by rraustad** [comment]
Oops, I didn't see your comment. My warning fix erased the previous CI results.

**Comment by rraustad** [comment]
@Myoldmopar merge in #10242 while your at it.

**Comment by rraustad** [comment]
FYI, the err diff (1 test warning) here is adding a `"` in the warning message that was previously missing at old line 6839/6840 and is now present at new line 6853. This warning does not make much sense.

    -   ** Warning ** ZoneHVAC:WaterToAirHeatPump = SPACE3-1 HEAT PUMP with Fan:SystemModel is used in  SPACE3-1 SUPPLY FAN"
    +   ** Warning ** ZoneHVAC:WaterToAirHeatPump = SPACE3-1 HEAT PUMP with Fan:SystemModel is used in "SPACE3-1 SUPPLY FAN"

**Comment by rraustad** [comment]
I ran the 23.1 defect file with results showing even more warnings than this branch:

    ************* EnergyPlus Warmup Error Summary. During Warmup: 0 Warning; 0 Severe Errors.
    ************* EnergyPlus Sizing Error Summary. During Sizing: 0 Warning; 0 Severe Errors.
    ************* EnergyPlus Completed Successfully-- 67168 Warning; 0 Severe Errors; Elapsed Time=00hr 00min  5.37sec

[in-23.1.idf.txt](https://github.com/NREL/EnergyPlus/files/12745477/in-23.1.idf.txt)

Used Chicago Ohare Intl Ap weather for these (and above) test results. Defect file shows Denver Intl Ap_CO_USA.

**Comment by Myoldmopar** [comment]
Thank you @rraustad!  I really appreciate the quick fix on this!  FYI @shorowit

**Pull request merged**